pkg/render/node.go

```diff
@@ -968,6 +968,10 @@ func (c *nodeComponent) nodeEnvVars() []v1.EnvVar {
 		}},
 	}
 
+	if c.cr.KubernetesProvider == operator.ProviderEKS {
```

Comment:
I think you'd want to check the CNI that's in use rather than the provider.  You can run EKS with Calico CNI or Kops with Amazon CNI, for example.  It's actually the CNI that does the wacky routing.